### PR TITLE
Process Restic stats output

### PR DESCRIPTION
**Pull request by DeepikaDixit**
## Change Overview

Process Restic stats output to get file count and size

## Pull request type

Please check the type of change your PR introduces:
- [ ] Work in Progress
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Trival/Minor
- [ ] Bugfix
- [X] Feature
- [ ] Documentation

## Issues

- #261 

## Test Plan

- [ ] Manual
- [X] Unit test

```
restic stats 7e17e764
repository c6746ff5 opened successfully, password is correct
found 2 old cache directories in /Users/deepikadixit/Library/Caches/restic, pass --cleanup-cache to remove them
scanning...
Stats for 7e17e764 in restore-size mode:
  Total File Count:   9
        Total Size:   10.322 KiB
```
- [ ] E2E

### CODE REVIEW

File: pkg/restic/restic.go [hunk 1]
```diff
@@ -229,3 +229,42 @@ func SnapshotIDFromBackupLog(output string) string {
...
+		if match2 != nil {
+			size = match2[0][1]
+		}
+	}
+	return SnapshotStatsModeFromStatsLog(output), fileCount, size
+}
+
+// SnapshotStatsModeFromStatsLog gets the Stats mode from Stats Command log
+func SnapshotStatsModeFromStatsLog(output string) string {
+	logs := regexp.MustCompile("[\n]").Split(output, -1)
```
Comment (tdmanv) [inline pkg/restic/restic.go]: Any reason you didn't use https://golang.org/pkg/strings/#Split ?

File: pkg/restic/restic.go [hunk 2]
```diff
@@ -229,3 +229,42 @@ func SnapshotIDFromBackupLog(output string) string {
...
+	for _, l := range logs {
+		// Log should contain "Total File Count:   xx"
```
Comment (pavannd1) [inline pkg/restic/restic.go]: [minor] update the comment

File: pkg/restic/restic.go [hunk 3]
```diff
@@ -229,3 +229,42 @@ func SnapshotIDFromBackupLog(output string) string {
...
+		pattern := regexp.MustCompile(`Stats for.*in\s+(.*?)\s+mode:`)
+		match := pattern.FindAllStringSubmatch(l, 1)
+		if match != nil {
+			return match[0][1]
```
Comment (tdmanv) [inline pkg/restic/restic.go]: are both these indices guaranteed to exist if `match != nil`?

File: pkg/restic/restic.go (hunk 2 above)
Comment (DeepikaDixit) [reply pkg/restic/restic.go]: Aaahh thanks!

File: pkg/restic/restic.go (hunk 1 above)
Comment (DeepikaDixit) [reply pkg/restic/restic.go]: Was trying to be consistent with other funcs in same file.

File: pkg/restic/restic.go (hunk 3 above)
Comment (DeepikaDixit) [reply pkg/restic/restic.go]: Based on testing yes. I will add a check

File: pkg/restic/restic.go [hunk 4]
```diff
@@ -229,3 +229,42 @@ func SnapshotIDFromBackupLog(output string) string {
...
+	}
+	var fileCount string
+	var size string
+	logs := regexp.MustCompile("[\n]").Split(output, -1)
+	for _, l := range logs {
+		// Log should contain "Total File Count:   xx"
+		pattern1 := regexp.MustCompile(`Total File Count:\s+(.*?)$`)
+		match1 := pattern1.FindAllStringSubmatch(l, 1)
+		if match1 != nil {
+			fileCount = match1[0][1]
```
[the remark this answers is not in the document]
Comment (DeepikaDixit) [reply pkg/restic/restic.go]: 👍

File: pkg/restic/restic.go [hunk 5]
```diff
@@ -229,3 +229,42 @@ func SnapshotIDFromBackupLog(output string) string {
...
+}
+
+// SnapshotStatsModeFromStatsLog gets the Stats mode from Stats Command log
+func SnapshotStatsModeFromStatsLog(output string) string {
+	logs := regexp.MustCompile("[\n]").Split(output, -1)
+	// Log should contain "Stats for .... in  xx mode"
+	pattern := regexp.MustCompile(`Stats for.*in\s+(.*?)\s+mode:`)
+	for _, l := range logs {
+		match := pattern.FindAllStringSubmatch(l, 1)
+		if match != nil && match[0] != nil {
```
Comment (tdmanv) [inline pkg/restic/restic.go]: This still doesn't check the second lookup. Same for the ones above as well

```suggestion
		if len(match) > 0 && len(match[0]) > 1 {
```

Comment (DeepikaDixit) [reply pkg/restic/restic.go]: Ahh okay. I will update and be more specific

**Pull request merged**